tools/IS_STAG/check_url.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import logging
+import os
+import sys
+import re
+
+_this_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, os.path.join(_this_dir, "../../src"))
+sys.path.insert(0, os.path.join(_this_dir, "../../src/pump"))
+
+import dspace  # noqa
+import settings  # noqa
+import project_settings  # noqa
+from utils import init_logging, update_settings  # noqa
+from _utils import read_json  # noqa: E402
+
+logging.getLogger("dspace.client").setLevel(logging.WARNING)
+_logger = logging.getLogger()
+
+env = update_settings(settings.env, project_settings.settings)
+init_logging(_logger, env["log_file"])
+
+
+class checker:
+    def __init__(self, dspace_be, data_path: str, new_key: str, cur_key: str):
+        """
+        Initialize the checker class with DSpace backend, data path, and URL keys.
+        """
+        self._dspace_be = dspace_be
+        self._data = read_json(data_path)
+        self._new_key = new_key
+        self._cur_key = cur_key
+        self._info = {
+            "invalid": []
+        }
+
+    @property
+    def info(self):
+        return self._info
+
+    def extract_uuid(self, url: str) -> str:
+        """
+        Extract UUID from the URL using a regex pattern.
+        """
+        pattern = r'bitstreams/([a-f0-9\-]$)'
```

Comment:
1. use rec, so that you do not need to recompile it every time
2. i do not know the url but unless it ends with uuid and without `/` then this is not correct

Reply:
and at least + is missing but better the correct number of chars

Reply:
Should I use corect number of char? Isn't it too complicated?
r'bitstreams/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})$' or
r'bitstreams/([a-f0-9\-]+)$'

Reply:
double check